src/Elliptic/Systems/Poisson/Equations.cpp

```diff
@@ -0,0 +1,182 @@
+    const tnsr::i<DataVector, Dim, Frame::Inertial>&
+        interface_unit_normal) noexcept {
+  get(*normal_dot_flux_for_field) =
+      -get<0>(interface_unit_normal) * get<0>(auxiliary_field);
+  for (size_t d = 1; d < Dim; d++) {
+    get(*normal_dot_flux_for_field) -=
+        interface_unit_normal.get(d) * auxiliary_field.get(d);
+  }
+
+  for (size_t d = 0; d < Dim; d++) {
+    normal_dot_flux_for_auxiliary_field->get(d) =
+        interface_unit_normal.get(d) * get(field);
+  }
+
+  // There is an additional minus sign in `lift_flux`
+  normal_dot_flux_for_field->get() *= -1.;
+  for (size_t d = 0; d < Dim; d++) {
+    normal_dot_flux_for_auxiliary_field->get(d) *= -1.;
+  }
+}
+
+template <size_t Dim>
+void InternalPenaltyFlux<Dim>::package_data(
+    gsl::not_null<Variables<package_tags>*> packaged_data,
+    const Scalar<DataVector>& field,
+    const tnsr::i<DataVector, Dim, Frame::Inertial>& grad_field,
+    const tnsr::i<DataVector, Dim, Frame::Inertial>& interface_unit_normal)
+    const noexcept {
+  get<LinearSolver::Tags::Operand<Field>>(*packaged_data) = field;
+
+  for (size_t d = 0; d < Dim; d++) {
+    get<NormalTimesFieldFlux>(*packaged_data).get(d) =
+        interface_unit_normal.get(d) * get(field);
+  }
+
+  get<NormalDotGradFieldFlux>(*packaged_data).get() =
+      get<0>(interface_unit_normal) * get<0>(grad_field);
+  for (size_t d = 1; d < Dim; d++) {
+    get<NormalDotGradFieldFlux>(*packaged_data).get() +=
+        interface_unit_normal.get(d) * grad_field.get(d);
+  }
+}
+
+template <size_t Dim>
+void InternalPenaltyFlux<Dim>::transform_boundary_data(
+    gsl::not_null<Variables<package_tags>*> packaged_data) const noexcept {
+  // This has its sign flipped to impose vanishing Dirichlet boundary conditions
+  get<LinearSolver::Tags::Operand<Field>>(*packaged_data).get() *= -1.;
+
+  // This must be flipped, but is canceled by the sign flip expected by
+  // operator()
+  //   for (size_t d = 0; d < Dim; d++) {
+  //   get<NormalTimesFieldFlux>(*packaged_data)
+  //       .get(d) *= -1.;
+  //   }
+
+  // For this we only perform the sign flip expected by `operator()`
+  get<NormalDotGradFieldFlux>(*packaged_data).get() *= -1.;
+}
+
+template <size_t Dim>
+void InternalPenaltyFlux<Dim>::operator()(
+    const gsl::not_null<Scalar<DataVector>*> numerical_flux_for_field,
+    const gsl::not_null<tnsr::I<DataVector, Dim, Frame::Inertial>*>
+        numerical_flux_for_auxiliary_field,
+    const Scalar<DataVector>& field_interior,
+    const tnsr::i<DataVector, Dim, Frame::Inertial>&
+        normal_times_field_interior,
+    const Scalar<DataVector>& normal_dot_grad_field_interior,
+    const Scalar<DataVector>& field_exterior,
+    const tnsr::i<DataVector, Dim, Frame::Inertial>&
+        minus_normal_times_field_exterior,
+    const Scalar<DataVector>& minus_normal_dot_grad_field_exterior) const
+    noexcept {
+  // Need polynomial degress and element size to compute this dynamically
```

Comment:
degrees    and what does element size mean?  physical size?  in what Frame?

Reply:
added a note to the struct doc